inventory.go

```diff
@@ -26,37 +51,27 @@ func main() {
 	if err != nil {
 		panic(fmt.Errorf("Fatal error config file: %s \n", err))
 	}
-	props := props{rackhdUrl: config.GetString("rackhd_api_url"), groups: config.GetStringSlice("groups")}
-
-	args := CLIArgs{}
-	arg.MustParse(&args)
-	if args.List {
-		handleList(props)
-		return
-	}
-
-	handleList(props)
-
-	return
+	return props{rackhdUrl: config.GetString("rackhd_api_url"), groups: config.GetStringSlice("groups"), filterGroup: config.GetString("filter_group")}
 }
 
-func handleList(props props) {
+// handleList returns a map with tags as keys and the list of hosts as values
+func handleList(props props) (map[string]interface{}, error){
+	// rackhdClient allows us to make calls to the REST API located at BaseUrl
 	rackhdClient := rackhd.Client{BaseUrl: props.rackhdUrl}
 
-	output := make(map[string]interface{})
 	hostvars := Hostvars{}
-
+	// ouput is used to return a map containing the host groups
```

Comment:
remove this, every time you feel the need to comment. always think how can you make your code more readable to show it's intent